--- docker-nc-entrypoint.old	2026-03-27 11:39:04.909244789 +0100
+++ docker-nc-entrypoint.new	2026-03-27 17:03:19.919603895 +0100
@@ -11,18 +11,10 @@
     [ -z "$(ls -A "$1/")" ]
 }
 
-run_as() {
-    if [ "$(id -u)" = 0 ]; then
-        su -p "$user" -s /bin/sh -c "$1"
-    else
-        sh -c "$1"
-    fi
-}
-
 # Execute all executable files in a given directory in alphanumeric order
 run_path() {
     local hook_folder_path="/docker-entrypoint-hooks.d/$1"
-    local found=0
+    local found=0 return_code
 
     echo "=> Searching for hook scripts (*.sh) to run, located in the folder \"${hook_folder_path}\""
 
@@ -41,11 +33,18 @@
 
             echo "==> Running the script (cwd: $(pwd)): \"${script_file_path}\""
             found=$((found+1))
-            run_as "${script_file_path}" || {
-                return_code="$?"
+
+            return_code=0
+            if [ "$(id -u)" = 0 ]; then
+                su -p "$user" -s /bin/sh "${script_file_path}" || return_code="$?"
+            else
+                "${script_file_path}" || return_code="$?"
+            fi
+
+            if [ "${return_code}" -ne "0" ]; then
                 echo "==> Failed at executing script \"${script_file_path}\". Exit code: ${return_code}"
                 exit 1
-            }
+            fi
 
             echo "==> Finished executing the script: \"${script_file_path}\""
         done
@@ -82,7 +81,7 @@
 }
 
 get_enabled_apps() {
-    run_as 'php /var/www/html/occ app:list' \
+    occ app:list \
         | sed -n '/^Enabled:$/,/^Disabled:$/p' \
         | sed '1d;$d' \
         | sed -n 's/^  - \([^:]*\):.*/\1/p' \
@@ -258,7 +257,7 @@
                         echo "Starting nextcloud installation"
                         max_retries=10
                         try=0
-                        until  [ "$try" -gt "$max_retries" ] || run_as "php /var/www/html/occ maintenance:install $install_options" 
+                        until [ "$try" -gt "$max_retries" ] || eval "occ maintenance:install $install_options"
                         do
                             echo "Retrying install..."
                             try=$((try+1))
@@ -274,7 +273,7 @@
                             NC_TRUSTED_DOMAIN_IDX=1
                             for DOMAIN in ${NEXTCLOUD_TRUSTED_DOMAINS}; do
                                 DOMAIN=$(echo "${DOMAIN}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')
-                                run_as "php /var/www/html/occ config:system:set trusted_domains $NC_TRUSTED_DOMAIN_IDX --value=\"${DOMAIN}\""
+                                occ config:system:set trusted_domains "$NC_TRUSTED_DOMAIN_IDX" --value="$DOMAIN"
                                 NC_TRUSTED_DOMAIN_IDX=$((NC_TRUSTED_DOMAIN_IDX+1))
                             done
 			    set +f # turn glob back on
@@ -292,7 +291,7 @@
             else
                 run_path pre-upgrade
 
-                run_as 'php /var/www/html/occ upgrade'
+                occ upgrade
 
                 get_enabled_apps > /tmp/list_after
                 disabled_apps="$(comm -23 /tmp/list_before /tmp/list_after || true)"
@@ -310,7 +309,7 @@
 
         # Update htaccess after init if requested
         if [ -n "${NEXTCLOUD_INIT_HTACCESS+x}" ] && [ "$installed_version" != "0.0.0.0" ]; then
-            run_as 'php /var/www/html/occ maintenance:update:htaccess'
+            occ maintenance:update:htaccess
         fi
     ) 9> /var/www/html/nextcloud-init-sync.lock
 
